backend/migration_tools/management/commands/migrate_to_v2.py

```diff
@@ -0,0 +1,367 @@
+            SELECT COUNT(*) FROM {self.migration_tracking_table}
+            WHERE migration_name = %s;"
+            """,
+            (migration_name,),
+        )
+        result = cur.fetchone()
+        cur.close()
+        conn.close()
+        return result[0] > 0
+
+    def record_migration(self, migration_name):
+        conn = self.connect_db(self.dest_db_config)
+        cur: cursor = conn.cursor()
+        cur.execute(
+            f"""
+            INSERT INTO {self.migration_tracking_table} (migration_name)
+            VALUES (%s);
+            """,
+            (migration_name,),
+        )
+        conn.commit()
+        cur.close()
+        conn.close()
+
+    def fetch_schema_names(
+        self, schemas_to_migrate: Optional[list[str]] = None
+    ) -> list[tuple[int, str]]:
+        conn = self.connect_db(self.src_db_config)
+        cur = conn.cursor()
+
+        # Process schemas_to_migrate: trim spaces and remove empty entries
+        if schemas_to_migrate:
+            schemas_to_migrate = [
+                schema.strip() for schema in schemas_to_migrate if schema.strip()
+            ]
+
+        if schemas_to_migrate:
+            placeholders = ",".join(["%s"] * len(schemas_to_migrate))
+            query = (
+                "SELECT id, schema_name FROM account_organization WHERE "
+                f"schema_name IN ({placeholders});"
+            )
+            cur.execute(query, schemas_to_migrate)
+        else:
+            query = "SELECT id, schema_name FROM account_organization;"
+            cur.execute(query)
+
+        schema_names = [(row[0], row[1]) for row in cur.fetchall()]
+        cur.close()
+        conn.close()
+        return schema_names
+
+    def migrate_single_row(
+        self,
+        row: tuple[Any, ...],
+        dest_cursor: cursor,
+        column_names: list[str],
+        new_key_transaction: dict[str, dict[str, Any]],
+    ) -> tuple[Any, ...]:
+        row = list(row)
+
+        for key, transaction in new_key_transaction.items():
+            if key in column_names:
+                query = transaction["query"]
+                params = transaction["params"]
+                param_values = [row[column_names.index(param)] for param in params]
+
+                dest_cursor.execute(query, param_values)
+                new_key_value = dest_cursor.fetchone()
+                if new_key_value:
+                    row[column_names.index(key)] = new_key_value[0]
+
+        for i, value in enumerate(row):
+            if isinstance(value, dict):
+                row[i] = json.dumps(value)
+        return tuple(row)
+
+    def migrate_rows(
+        self,
+        migration_name: str,
+        src_cursor: cursor,
+        dest_cursor: cursor,
+        dest_conn: connection,
+        dest_query: str,
+        column_names: list[str],
+        new_key_transaction: dict[str, dict[str, Any]],
+    ) -> None:
+        batch_count = 0
+        total_rows = 0
+
+        while True:
```

Comment:
@muhammad-ali-e while True are dangerous. We can go in to infinite loop. My suggestion use the rows variable to do this. 

Do a `rows = src_cursor.fetchmany(self.batch_size)` in the beginning before staring loop. Then make the condition as `while rows`. And similarly do a `rows = src_cursor.fetchmany(self.batch_size)` at end of the loop.

Reply:
let me check

Reply:
addressed